zipline-bytecode/src/main/kotlin/app/cash/zipline/bytecode/lineNumbers.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2021 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.zipline.bytecode
+
+import okio.BufferedSink
+import okio.BufferedSource
+import okio.Closeable
+
+/** Decode the pc2line table. */
+class LineNumberReader(
+  functionLineNumber: Int,
+  private val source: BufferedSource,
+) : Closeable by source {
+  var pc: Int = 0
```

Comment:
Also, why are these public, should be private?

Reply:
I was hoping to avoid allocating a `Pair<Int, Int>` and/or a 2-element `IntArray`. If you’d like to try stuff here too, I’m game.